### ISSUE TITLE
Error processing existing log groups

### ISSUE THREAD

**Issue by drummerjoe** [lumigo-io/SAR-Logging#14]
I'm seeing the following when trying to run this against some existing log groups.  I installed this from the Serverless Repo and only configured the ARN, prefix, and override manual configs.  It appears to work properly until it actually tries to subscribe.  It successfully deletes existing subscriptions, etc.

```
"message": "cannot process existing log group, skipped...",
    "logGroupName": "/aws/lambda/prod-redemption-api",
    "errorName": "InvalidParameterType",
    "errorMessage": "Expected params.logGroupName to be a string",
    "stackTrace": "InvalidParameterType: Expected params.logGroupName to be a string\n    at ParamValidator.fail (/var/task/node_modules/aws-sdk/lib/param_validator.js:50:37)\n    at ParamValidator.validateType (/var/task/node_modules/aws-sdk/lib/param_validator.js:232:10)\n    at ParamValidator.validateString (/var/task/node_modules/aws-sdk/lib/param_validator.js:154:32)\n    at ParamValidator.validateScalar (/var/task/node_modules/aws-sdk/lib/param_validator.js:130:21)\n    at ParamValidator.validateMember (/var/task/node_modules/aws-sdk/lib/param_validator.js:94:21)\n    at ParamValidator.validateStructure (/var/task/node_modules/aws-sdk/lib/param_validator.js:75:14)\n    at ParamValidator.validateMember (/var/task/node_modules/aws-sdk/lib/param_validator.js:88:21)\n    at ParamValidator.validate (/var/task/node_modules/aws-sdk/lib/param_validator.js:34:10)\n    at Request.VALIDATE_PARAMETERS (/var/task/node_modules/aws-sdk/lib/event_listeners.js:126:42)\n    at Request.callListeners (/var/task/node_modules/aws-sdk/lib/sequential_executor.js:106:20)",

```

**Comment by theburningmonk** [comment]
@drummerjoe was it able to process most of the existing groups? or did it fail on ALL of them?

**Comment by drummerjoe** [comment]
Failed on all.  Here's the output from the first group it tried to process:

```
"message": "failed to subscribe log group",
    "logGroupName": [
        "/aws/lambda/prod-analytic-api",
        "/aws/lambda/prod-athena-scheduled-load-partitions",
        "/aws/lambda/prod-auth-api",
        "/aws/lambda/prod-automation-post-api",
        "/aws/lambda/prod-category-api",
        "/aws/lambda/prod-cognito-custom-message-api",
        "/aws/lambda/prod-es-index-api",
        "/aws/lambda/prod-image-api",
        "/aws/lambda/prod-location-api",
        "/aws/lambda/prod-notification-api",
        "/aws/lambda/prod-offer-api",
        "/aws/lambda/prod-payment-api",
        "/aws/lambda/prod-payment-processing",
        "/aws/lambda/prod-post-api",
        "/aws/lambda/prod-redemption-api",
        "/aws/lambda/prod-scheduled-stats-api",
        "/aws/lambda/prod-total-income-api",
        "/aws/lambda/prod-user-api"
    ],
    "errorName": "InvalidParameterType",
    "errorMessage": "Expected params.logGroupName to be a string",
    "stackTrace": "InvalidParameterType: Expected params.logGroupName to be a string\n    at ParamValidator.fail (/var/task/node_modules/aws-sdk/lib/param_validator.js:50:37)\n    at ParamValidator.validateType (/var/task/node_modules/aws-sdk/lib/param_validator.js:232:10)\n    at ParamValidator.validateString (/var/task/node_modules/aws-sdk/lib/param_validator.js:154:32)\n    at ParamValidator.validateScalar (/var/task/node_modules/aws-sdk/lib/param_validator.js:130:21)\n    at ParamValidator.validateMember (/var/task/node_modules/aws-sdk/lib/param_validator.js:94:21)\n    at ParamValidator.validateStructure (/var/task/node_modules/aws-sdk/lib/param_validator.js:75:14)\n    at ParamValidator.validateMember (/var/task/node_modules/aws-sdk/lib/param_validator.js:88:21)\n    at ParamValidator.validate (/var/task/node_modules/aws-sdk/lib/param_validator.js:34:10)\n    at Request.VALIDATE_PARAMETERS (/var/task/node_modules/aws-sdk/lib/event_listeners.js:126:42)\n    at Request.callListeners (/var/task/node_modules/aws-sdk/lib/sequential_executor.js:106:20)",
    "awsRegion": "us-east-1",
    "functionName": "serverlessrepo-devops-aut-SubscribeExistingLogGrou-WW25ENAE8QAS",
    "functionVersion": "$LATEST",
    "functionMemorySize": "128",
    "level": 50,
    "sLevel": "ERROR"
```

**Comment by theburningmonk** [comment]
mm.. weird, that looks as though it was trying to use the array
```
[
        "/aws/lambda/prod-analytic-api",
        "/aws/lambda/prod-athena-scheduled-load-partitions",
        "/aws/lambda/prod-auth-api",
        "/aws/lambda/prod-automation-post-api",
        "/aws/lambda/prod-category-api",
        "/aws/lambda/prod-cognito-custom-message-api",
        "/aws/lambda/prod-es-index-api",
        "/aws/lambda/prod-image-api",
        "/aws/lambda/prod-location-api",
        "/aws/lambda/prod-notification-api",
        "/aws/lambda/prod-offer-api",
        "/aws/lambda/prod-payment-api",
        "/aws/lambda/prod-payment-processing",
        "/aws/lambda/prod-post-api",
        "/aws/lambda/prod-redemption-api",
        "/aws/lambda/prod-scheduled-stats-api",
        "/aws/lambda/prod-total-income-api",
        "/aws/lambda/prod-user-api"
    ]
```
as the log group name parameter when subscribing!

mm... I don't understand why it'd do that. So you were deploying the app from the Serverless repo console? Can you paste the configuration options you used here?

**Comment by drummerjoe** [comment]
Yeah confusing - looks like a big array.  Here's the parameters:

```
DestinationArn | arn:aws:lambda:us-east-1:XXX:function:LogsToElasticsearch_devops-logdepot | -
-- | -- | --
ExcludePrefix | - | -
ExcludeTags | - | -
ExcludeTagsMode | OR | -
FilterName | ship-logs | -
FilterPattern | [timestamp=*Z, request_id="*-*", event] | -
OverrideManualConfigs | true | -
Prefix | /aws/lambda/prod- | -
Tags | - | -
TagsMode | OR

```

**Comment by drummerjoe** [comment]
One thing I just noticed is that even though I left filtername and filterpattern blank, looks like it might have included them for some reason - can't be sure.  FWIW the Lambda does appear to be attempting to process the correct log groups (/aws/lambda/prod-).

**Comment by theburningmonk** [comment]
hi @drummerjoe I finally found the issue, it should be fixed in v1.9.3
